docs/quickstart.md

```diff
@@ -55,94 +55,81 @@ sudo cp target/x86_64-unknown-linux-musl/release/{firecracker,jailer} /usr/local
+echo "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable" | \
+     sudo tee /etc/apt/sources.list.d/docker.list
+sudo DEBIAN_FRONTEND=noninteractive apt-get update
+sudo DEBIAN_FRONTEND=noninteractive apt-get \
+     install --yes \
+     docker-ce
+sudo usermod -aG docker $(whoami)
+exec newgrp docker
 
 cd ~
 
-# Check out firecracker-containerd and build it
+# Check out firecracker-containerd and build it.  This includes:
+# * block-device snapshotter gRPC proxy plugins
+# * firecracker-containerd runtime, a containerd v2 runtime
+# * firecracker-containerd agent, an inside-VM component
+# * runc, to run containers inside the VM
+# * a Debian-based root filesystem configured as read-only with a read-write
+#   overlay
+# * firecracker-containerd, an alternative containerd binary that includes the
+#   firecracker VM lifecycle plugin and API
 git clone https://github.com/firecracker-microvm/firecracker-containerd.git
 cd firecracker-containerd
+git submodule update --init --recursive
 sudo DEBIAN_FRONTEND=noninteractive apt-get install -y dmsetup
-make STATIC_AGENT='true'
-sudo cp runtime/containerd-shim-aws-firecracker snapshotter/cmd/{devmapper/devmapper_snapshotter,naive/naive_snapshotter} /usr/local/bin
+make all image
+sudo make install
 
 cd ~
 
-# Download kernel and generic VM image
+# Download kernel
 curl -fsSL -o hello-vmlinux.bin https://s3.amazonaws.com/spec.ccfc.min/img/hello/kernel/hello-vmlinux.bin
-curl -fsSL -o hello-rootfs.ext4 https://s3.amazonaws.com/spec.ccfc.min/img/hello/fsfiles/hello-rootfs.ext4
-
-# Inject the agent, runc, and a startup script into the VM image
-mkdir /tmp/mnt
-# Construct fc-agent.start
-cat >fc-agent.start <<EOF
-#!/bin/sh
-mkdir -p /container
-exec > /container/agent-debug.log # Debug logs from the agent
-exec 2>&1
-touch /container/runtime
-mkdir /container/rootfs
-mount -t auto -o rw /dev/vdb /container/rootfs
-cd /container
-/usr/local/bin/agent -id 1 -debug &
-EOF
-chmod +x fc-agent.start
-truncate --size=+50M hello-rootfs.ext4
-/sbin/e2fsck -f hello-rootfs.ext4
-/sbin/resize2fs hello-rootfs.ext4
-sudo mount hello-rootfs.ext4 /tmp/mnt
-sudo cp $(which runc) firecracker-containerd/agent/agent /tmp/mnt/usr/local/bin
-sudo cp fc-agent.start /tmp/mnt/etc/local.d
-sudo ln -s /etc/init.d/local /tmp/mnt/etc/runlevels/default/local
-sudo ln -s /etc/init.d/cgroups /tmp/mnt/etc/runlevels/default/cgroups
-sudo umount /tmp/mnt
-rmdir /tmp/mnt
 
-cd ~
-
-# Configure containerd to use our new snapshotter
-sudo mkdir -p /etc/containerd
-sudo tee -a /etc/containerd/config.toml <<EOF
+# Configure our firecracker-containerd binary to use our new snapshotter and
+# separate storage from the default containerd binary
+sudo mkdir -p /etc/firecracker-containerd
+sudo mkdir -p /var/lib/firecracker-containerd/containerd
+sudo mkdir -p /run/firecracker-containerd
+sudo tee /etc/firecracker-containerd/config.toml <<EOF
+disabled_plugins = ["cri"]
```

Comment:
Did we ever figure out why we needed this? Is it still needed, or are we just cargo-culting it?

Reply:
Looks like it was unnecessary.